### PR TITLE
Update run_tests.yaml

### PR DESCRIPTION
**Pull request by zalbiraw**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). If PRing from your fork, don't come from your `master`!
- [ ] Make sure you are making a pull request against our **`master` branch** (left side). Also, it would be best if you started *your change* off *our latest `master`*.
- [ ] Make sure you are updating [CHANGELOG.md](https://github.com/TykTechnologies/tyk-operator/blob/master/CHANGELOG.md) based on your changes.
- [ ] My change requires a change to the documentation.
  - [ ] If you've changed APIs, describe what needs to be updated in the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] If you've changed API models, please update CRDs.
  - [ ] `make manifests`
  - [ ] `make helm`
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] Check your code additions will not fail linting checks:
  - [ ] `gofmt -s -w .`
  - [ ] `go vet ./...`
  - [ ] `golangci-lint run`

### CODE REVIEW

**Comment by zalbiraw** [comment]
Hi @komalsukhani this was brought up by @asoorm and @caroltyk. The idea is that we should be testing against at least the major tyk versions in case of breaking changes. Those major branches get updated with every minor release. Example:

4.0 -> 4.0.10
4.3 -> 4.3.1
etc

Since nightly does not encompass all the changes and will never detect merging issues that may happen with already existing branches. Its an added layer of defence.

**Comment by komalsukhani** [comment]
Hi @zalbiraw, We have agreed to run test against `3.2, 4.3, 4.0` tyk versions. Can you update the PR accordingly?

File: .github/workflows/run_tests.yaml [hunk 1]
```diff
@@ -39,7 +39,7 @@ jobs:
       fail-fast: false
       matrix: 
         k8s-version: ${{ fromJSON(needs.matrix-generator.outputs.k8s-matrix) }}
-        tyk-version: ['v3.2', 'v4.0']
+        tyk-version: ['v3.2', 'v4.0', 'v4.1', 'v4.2', 'v4.3']
```
Comment (zalbiraw) [inline .github/workflows/run_tests.yaml:42]:
```suggestion
        tyk-version: ['v3.2', 'v4.0', 'v4.3']
```

**Comment by zalbiraw** [comment]
Done!

**Review by komalsukhani — APPROVED** [review]

**Comment by komalsukhani** [comment]
Thank you @zalbiraw

**Comment by zalbiraw** [comment]
Anytime!

**Pull request merged**